src/main/java/com/frameworkium/core/ui/element/AbstractStreamTable.java

```diff
@@ -217,13 +212,13 @@
      * This shouldn't be needed outside this class.
      * If you think it should be, raise a bug to discuss.
      */
-    private int getHeaderIndex(Predicate<WebElement> headerPredicate) {
-        List<WebElement> headings = getHeadings().collect(toList());
-        return IntStream.range(0, headings.size())
-                .filter(i -> headerPredicate.test(headings.get(i)))
+    private long getHeaderIndex(Predicate<WebElement> headerPredicate) {
+        return Streams.mapWithIndex(
+                getHeadings(), (we, i) -> headerPredicate.test(we) ? i : null)
+                .filter(Objects::nonNull)
                 .findFirst()
                 .orElseThrow(() -> new NoSuchElementException(
-                        "Table header " + headerPredicate + " not found"));
+                        "No header matches " + headerPredicate));
```

Comment:
I just realized that headerPredicate returns as a Java Object representation and not a String representation of the lambda function. 
Googling suggests there is no direct way to stringify it